src/ts/src/common/components/library/modal/index.tsx

```diff
@@ -35,8 +36,9 @@ const Modal: FunctionComponent<Props> = ({
     }
 
     const buttons = buttonInfo.map((button, index) => {
-      let link: string | undefined = button.link;
-      let onClick: (() => void) | undefined = undefined;
+      let link = button.link;
+      let onClick;
+      const Button = button.Button || DefaultButton;
```

Comment:
Modal component now accepts an optional `button.Button` component, so we can use that custom component to fetch presigned URL and render the link accordingly